enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultNamespaceEnricher.java

```diff
@@ -0,0 +1,298 @@
+        config = new ResourceConfig.Builder(xmlResourceConfig)
+                .build();
+
+        handlerHub = new HandlerHub(
+                getContext().getGav(), getContext().getConfiguration().getProperties());
+    }
+
+    /**
+     * This method will create a default Namespace or Project if a namespace property is
+     * specified in the xml resourceConfig or as a parameter to a mojo.
+     * @param platformMode
+     * @param builder
+     */
+    @Override
+    public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
+
+        final String name = config.getNamespace();
+
+        if (name == null || name.length() == 0) {
+            return;
+        }
+
+        if (!KubernetesResourceUtil.checkForKind(builder, NAMESPACE_KINDS)) {
+            String type = getConfig(Config.type);
+            if ("project".equalsIgnoreCase(type) || "namespace".equalsIgnoreCase(type)) {
+                if (platformMode == PlatformMode.kubernetes) {
+                    try {
+                        createKubernetesClient().namespaces().list().getItems().forEach(namespace -> {
+                            if(namespace.getMetadata().getName().equals(name)) {
+                                log.warn("A namespace with name: %s already exists in your cluster.", namespace.getMetadata().getName());
+                            }
+                        });
+
+                    } catch (KubernetesClientException e) {
+                        log.info("Could not connect to a cluster.");
+                    }
+
+                    log.info("Adding a default Namespace");
+                    Namespace namespace = handlerHub.getNamespaceHandler().getNamespace(config);
+                    builder.addToNamespaceItems(namespace);
+                } else {
+                    try {
+                        createOpenShiftClient().projects().list().getItems().forEach(project -> {
+                            if (project.getMetadata().getName().equals(name)) {
+                                log.warn("A project with name: %s already exists in your cluster.", project.getMetadata().getName());
+                            }
+                        });
+                    } catch (KubernetesClientException e) {
+                        log.info("Could not connect to a cluster.");
+                    }
+
+                    log.info("Adding a default Project");
+                    Project project = handlerHub.getProjectHandler().getProject(config);
+                    builder.addToProjectItems(project);
+                }
+            }
+        }
+    }
+
+    /**
+     * This method will annotate all the items in the KubernetesListBuilder with the
+     * created new namespace or project.
+     * @param platformMode
+     * @param builder
+     */
+    @Override
+    public void enrich(PlatformMode platformMode, KubernetesListBuilder builder) {
+
+        builder.accept(new TypedVisitor<DeploymentBuilder>() {
```

Comment:
We're only editing metadata, I wonder if we can just iterate over ObjectMetaBuilder rather than iterating over all types individually.

Reply:
good catch, thanks !

I tried with a visitor on HasMetaData first, that didn't work. I think it works on builder classes only